File: example/ios/ExampleAuthProvider.swift
//
//  ExampleAuthProvider.swift
//  example
//
//  Created by Adam Mihalik on 25/02/2023.
//  Copyright © 2023 Facebook. All rights reserved.
//

import Foundation
import ExponeaSDK

@objc(ExponeaAuthProvider)
public class ExampleAuthProvider: NSObject, AuthorizationProviderType {
    required public override init() { }
    public func getAuthorizationToken() -> String? {
        // Receive and return JWT token here.
        return CustomerTokenStorage.shared.retrieveJwtToken()
        // NULL as returned value will be handled by SDK as 'no value'
    }
}

File: example/android/app/src/main/java/com/exponea/example/services/ExampleAuthProvider.java
package com.exponea.example.services;

import androidx.annotation.Nullable;
import com.exponea.RNAuthorizationProvider;
import com.exponea.example.manager.CustomerTokenStorage;

public class ExampleAuthProvider implements RNAuthorizationProvider {
    @Nullable
    @Override
    public String getAuthorizationToken() {
        // Receive and return JWT token here.
        return CustomerTokenStorage.INSTANCE_HOLDER.INSTANCE.retrieveJwtToken();
        // NULL as returned value will be handled by SDK as 'no value' and leads to Error
    }
}
